stablehlo/reference/Element.cpp

```diff
@@ -598,6 +604,210 @@ Element atan2(const Element &e1, const Element &e2) {
                                      debugString(type).c_str()));
 }
 
+Element bitcastConvertManyToOne(Type type, ArrayRef<Element> elements) {
```

Comment:
IMO, I don't think we need to have these three functions. For example, we can just copy the storage of operand to target at _byte granularity_. That is also followed from what we have in the spec.

With that we do not have to deal with interpreting the storage with a particular type and and write code specific to that type.

Reply:
There are two problems with this approach:
1. Some element types like int4 or booleans do not take up full 8 bits. The remaining bits are unset (could be anything), so by reinterpreting them as smaller or larger bits, it would result in wrong interpretation of the bits. E.g.
```
8x booleans -> 1x int8
little endian representation -> binary
Expected:
false, true, false, true, false, false, true, true -> -54
1111000[1] 1111000[1] 1111000[0] 1111000[0] 1111000[1] 1111000[0] 1111000[1] 1111000[0] -> 11001010

Actual (the actual representation may differ but the idea is same):
false, true, false, true, false, false, true, true -> -16
11110001 11110001 11110000 11110000 11110001 11110000 11110001 [11110000] -> 11110000
```
and even if we go with this approach, we'd still need to specialize booleans and int4/uint4 types + handle the three types of conversions separately, so there won't be any simplifications made.

2. The implementation does not align with the spec. I've updated the PR to not expose the three functions in `Ops.cpp` and put them in `Elements.cpp` as helper functions, which makes the implementation in `Ops.cpp` closer to the spec.

Reply:
agreed. Thanks for explaining.